lib/WlansvcOperationHandler.hpp

```diff
@@ -18,6 +18,11 @@ class WlansvcOperationHandler : public OperationHandler
     WlansvcOperationHandler(ProxyWifiCallbacks callbacks, std::vector<std::unique_ptr<IWlanInterface>> wlanInterfaces, std::shared_ptr<Wlansvc::WlanApiWrapper>& wlansvc)
         : OperationHandler{std::move(callbacks), std::move(wlanInterfaces)}, m_wlansvc{wlansvc}
     {
+        if (!m_wlansvc)
```

Comment:
I agree with this change in isolation since it fixes a definite bug.

However, on a broader scale, does it make sense to allow an instance of this to be created with `wlansvc == nullptr`? It seems the instance will do nothing in that case.

Should there be additional changes in other parts of the code that create an instance of this? For example, `MakeWlansvcOperationHandler` is one place that creates a `WlansvcOperationHandler` instance and it already checks the `std::shared_ptr<Wlansvc::WlanApiWrapper>` it passes here. Does it make sense to continue when the pointer is `nullptr`?

Reply:
Since I have to pass a pointer to wlansvc, it felt the check is needed anyway.
At creation time, we could decide to instantiate an `OperationHandler` which doesn't need `wlansvc` instead of a `WlansvcOperationHandler`, this would be strictly equivalent.

We need to continue when `wlansvc` is null because we still want to support the interface for client provided SSIDs.

Reply:
Created #2 to track future consideration of this. Will approve the PR as it is since I agree, the `nullptr` check is prudent here anyway.